contributions/authority/auth0/src/main/java/com/yahoo/athenz/auth/oauth/auth0/Auth0Jwt.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2020 Yahoo Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yahoo.athenz.auth.oauth.auth0;
+
+import java.util.LinkedHashMap;
+import com.yahoo.athenz.auth.AuthorityConsts;
+import com.yahoo.athenz.auth.oauth.token.DefaultOAuthJwtAccessToken;
+import com.yahoo.athenz.auth.oauth.token.OAuthJwtAccessToken;
+import com.yahoo.athenz.auth.util.AthenzUtils;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jws;
+import io.jsonwebtoken.RequiredTypeException;
+
+/**
+ * Custom for Auth0 JWT access token format
+ */
+public class Auth0Jwt extends DefaultOAuthJwtAccessToken {
+
+    private static String claimClientId = "https://myapp.example.com/client_id";
+    private static String claimConfirm = "https://myapp.example.com/cnf";
+    private static String userDomain = "user";
+
+    /**
+     * Create Auth0 JWT access token object
+     * @param  jws JWS claims
+     */
+    public Auth0Jwt(Jws<Claims> jws) {
+        super(jws);
+    }
+
+    @Override
+    public String getSubject() {
+        String subject = this.body.getSubject();
+        if (subject == null) {
+            return null;
+        }
+        // normalize as Athenz principal
+        return (AthenzUtils.getPrincipalName(Auth0Jwt.userDomain, subject.replace('|', '-'))).toLowerCase();
```

Comment:
I would suggest if we have a AthenzUtils.getPrincipal method then that method should already return values in lowercase since that's what Athenz uses everywhere. Then we don't have to remember to call toLowerCase() everywhere.

Reply:
Thank you for suggesting.
I updated both util methods in 646436c8 & f8cb9ade.